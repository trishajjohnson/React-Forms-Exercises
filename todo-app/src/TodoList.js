import React, { useState } from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    const [todos, setTodos] = useState([]);

    const addTodo = todo => {
        setTodos(todos => [...todos, {...todo}]);
    }
    
    const removeTodo = id => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    };
    
    const allTodos = todos.map(todo => (
        <Todo 
            key = {todo.id}
            id = {todo.id}
            task = {todo.task}
            handleRemove = {removeTodo}
        />
    ));
    
    return (
        <div className="Todolist">
            <NewTodoForm createTodo={addTodo}/>
            {allTodos}
        </div>
    );
}

export default TodoList;