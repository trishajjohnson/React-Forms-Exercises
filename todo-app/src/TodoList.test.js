import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

// Smoke test for TodoList component
it("should render without crashing", function() {
  render(<TodoList />)
});

// Snapshot test for TodoList component
it("should match snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add new todo", function() {
    const {queryByText, getByLabelText} = render(<TodoList />);
    expect(queryByText("Wash Car")).not.toBeInTheDocument();

    const input = getByLabelText("Task");
    const btn = queryByText("Add Todo");
    fireEvent.change(input, { target: { value: "Wash Car" } });
    fireEvent.click(btn);
    expect(queryByText("Wash Car")).toBeInTheDocument();

});

it("should delete todo", function() {
    const {queryByText, getByLabelText} = render(<TodoList />);
    const input = getByLabelText("Task");
    const btn = queryByText("Add Todo");
    fireEvent.change(input, { target: { value: "Wash Car" } });
    fireEvent.click(btn);
    expect(queryByText("Wash Car")).toBeInTheDocument();

    const deleteBtn = queryByText("X");
    fireEvent.click(deleteBtn);
    expect(queryByText("Wash Car")).not.toBeInTheDocument();

});