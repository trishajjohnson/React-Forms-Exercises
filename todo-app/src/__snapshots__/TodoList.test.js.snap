// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todolist"
  >
    <form>
      <label
        for="task"
      >
        Task
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
